File: 4/b/main.js
// Function executing on each iteration
var f = (x) => Math.sqrt(Math.pow(x, 3));

const range = [0, 4];

var getLength = (func, iterations = 10) => {
  var step = (range[1] - range[0]) / iterations;
  var prev = 0;
  var sum = 0;

  for (let i = 1; i <= iterations; i++) {
    let value = f(step * i);
    let height = value - prev;
    let hypotenuse = Math.sqrt(Math.pow(step, 2) + Math.pow(height, 2));
    prev = value;
    sum += hypotenuse;
  }

  return sum;
};


var oc = 9.073;//89388
var i = 1;
var res;

do {
  res = getLength(f, i);
  console.log(res, oc);
  i++;
} while(res < oc || i < 50);

File: 4/b/main.pas
program main;

{$mode objfpc}{$H+}

uses math;

function getLength(iterations: integer): double;
var
  i, x: integer;
  val, prev, hypotenuse, sum, step, height: double;
begin
  x := 4;
  prev := 0;
  step := x / iterations;
  sum := 0;

  for i:=1 to iterations do
  begin
    val := sqrt(power(step * i, 3));
    height := val - prev;
    prev := val;
    hypotenuse := sqrt(power(step, 2) + power(height, 2));
    sum := sum + hypotenuse;
  end;
  result := sum;
end;


var
  result, sample: double;
  i: integer;
begin

  // 9,073415289388
  sample := 9.073;
  i := 1;

  repeat
    result := getLength(i);
    writeLn('Iteration ', i, ' result ', result);
    inc(i);
  until result > sample;

end.

